source-hubspot-native/source_hubspot_native/resources.py

```diff
@@ -22,56 +25,344 @@
+        crm_object_streamed_no_batch(SubscriptionChanges, http, fetch_recent_subscription_changes),
+        crm_object_streamed_no_batch(EmailEvents, http, fetch_recent_email_events),
+        crm_object_streamed_no_batch(TicketPipelines, http, fetch_recent_ticket_pipelines),
+        crm_object_streamed_no_batch(DealPipelines, http, fetch_recent_deal_pipelines),
+        crm_object_paginated(EngagementCalls, http),
+        crm_object_paginated(EngagementEmails, http),
+        crm_object_paginated(EngagementMeetings, http),
+        crm_object_paginated(EngagementNotes, http),
+        crm_object_paginated(EngagementTasks, http),
+        crm_object_paginated(Goals, http),
+        crm_object_paginated(FeedbackSubmissions, http),
+        crm_object_paginated(LineItems, http),
+        crm_object_paginated(Products, http),
+        crm_object_paginated(Ticket, http),
+        crm_object_custom(MarketingEmails, http, fetch_recent_marketing_emails),
+        subscription_object(EmailSubscriptions, http, fetch_email_subscriptions),
+        crm_object_custom(MarketingForms, http, fetch_marketing_forms),
+        crm_object_custom(Owners, http, fetch_owners),
+        workflow_object(Workflows, http, fetch_workflows),
         properties(http),
     ]
 
+    all_custom_objects = await fetch_custom_objects(log, http)
+    if len(all_custom_objects["results"]) != 0:
+        for objects in all_custom_objects["results"]:
+            BaseCRMObject.NAME = objects["labels"]["plural"].lower()
+            BaseCRMObject.PRIMARY_KEY = ["/id"]
+            BaseCRMObject.PROPERTY_SEARCH_NAME = objects["labels"]["plural"].lower()
+            BaseCRMObject.ASSOCIATED_ENTITIES = []
+
+            resources_list.append(custom_objects(BaseCRMObject, http))
+
+    return resources_list
+
+
+def crm_object_paginated(
+    cls: type[CRMObject], http: HTTPSession
+) -> common.Resource:
+    """Base Resource to run V3 API objects using pagination
+
+    Args:
+        cls (type[CRMObject]): _description_
+        http (HTTPSession): _description_
+        fetch_recent (FetchRecentFn): _description_
+
+    Returns:
+        common.Resource: _description_
+    """
+    def open(
+        binding: CaptureBinding[ResourceConfig],
+        binding_index: int,
+        state: ResourceState,
+        task: Task,
+    ):
+        common.open_binding(
+            binding,
+            binding_index,
+            state,
+            task,
+            fetch_page=functools.partial(fetch_page, cls, http),
+        )
+
+    started_at = datetime.now(tz=UTC)
+
+    return common.Resource(
+        name=cls.NAME,
+        key=cls.PRIMARY_KEY,
+        model=cls,
+        open=open,
+        initial_state=ResourceState(
+            inc=ResourceState.Incremental(cursor=started_at),
+            backfill=ResourceState.Backfill(next_page=None, cutoff=started_at),
+        ),
+        initial_config=ResourceConfig(name=cls.NAME, interval=timedelta(minutes=7)),
```

Comment:
I asked about this elsewhere in a snapshot output as well - I'd like to know more about the default time delta of 7 minutes.

Reply:
I believe to have answered that on one of the more initial comments, please tell me if that explanation is enough